aws/resource_aws_waf_geo_match_set.go

```diff
@@ -0,0 +1,201 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/waf"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsWafGeoMatchSet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWafGeoMatchSetCreate,
+		Read:   resourceAwsWafGeoMatchSetRead,
+		Update: resourceAwsWafGeoMatchSetUpdate,
+		Delete: resourceAwsWafGeoMatchSetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"geo_match_constraints": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsWafGeoMatchSetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafconn
+
+	log.Printf("[INFO] Creating GeoMatchSet: %s", d.Get("name").(string))
+
+	wr := newWafRetryer(conn, "global")
+	out, err := wr.RetryWithToken(func(token *string) (interface{}, error) {
+		params := &waf.CreateGeoMatchSetInput{
+			ChangeToken: token,
+			Name:        aws.String(d.Get("name").(string)),
+		}
+
+		return conn.CreateGeoMatchSet(params)
+	})
+	if err != nil {
+		return errwrap.Wrapf("[ERROR] Error creating GeoMatchSet: {{err}}", err)
+	}
+	resp := out.(*waf.CreateGeoMatchSetOutput)
+
+	d.SetId(*resp.GeoMatchSet.GeoMatchSetId)
+
+	return resourceAwsWafGeoMatchSetUpdate(d, meta)
+}
+
+func resourceAwsWafGeoMatchSetRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafconn
+	log.Printf("[INFO] Reading GeoMatchSet: %s", d.Get("name").(string))
+	params := &waf.GetGeoMatchSetInput{
+		GeoMatchSetId: aws.String(d.Id()),
+	}
+
+	resp, err := conn.GetGeoMatchSet(params)
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "WAFNonexistentItemException" {
```

Comment:
Do you mind using a helper function here and simplify the code that way a bit? 😉 

```go
if isAWSErr(err, "WAFNonexistentItemException", "") {
```